ui/src/app/machines/components/MachineHeaderForms/ActionFormWrapper/ActionFormWrapper.tsx

```diff
@@ -81,141 +86,142 @@ const getErrorSentence = (action: MachineActions, count: number) => {
+    // Cloning in the UI works inverse to the rest of the machine actions - we
+    // select the destination machines first, then select the machine to perform
+    // the clone action, so we don't care what the selected machines are here.
+    return statusEntries.reduce<number>(
+      (count, [, statuses]) => (statuses["cloning"] ? count + 1 : count),
+      0
+    );
+  }
+  return statusEntries.reduce<number>((count, [machineID, statuses]) => {
+    const machineInSelection = machines.some(
+      (machine) => machine.system_id === machineID
+    );
+    const machineIsProcessing = statuses[actionStatus] === true;
+    if (machineInSelection && machineIsProcessing) {
+      return count + 1;
+    }
+    return count;
+  }, 0);
 };
 
 export const ActionFormWrapper = ({
   action,
+  applyConfiguredNetworking,
   clearHeaderContent,
-  headerContent,
+  hardwareType,
+  machines,
   setSearchFilter,
   viewingDetails,
 }: Props): JSX.Element => {
   const dispatch = useDispatch();
   const onRenderRef = useScrollOnRender<HTMLDivElement>();
-  const activeMachineId = useSelector(machineSelectors.activeID);
-  const { machinesToAction, processingCount } = useMachineActionForm(action);
-  const actionableMachineIDs = machinesToAction.reduce<
-    Machine[MachineMeta.PK][]
-  >((machineIDs, machine) => {
-    if (canOpenActionForm(machine, action)) {
-      machineIDs.push(machine.system_id);
-    }
-    return machineIDs;
-  }, []);
-  // The action should be disabled if not all the selected machines can perform
-  // the selected action. When machines are processing the available actions
-  // can change, so the action should not be disabled while processing.
-  const actionDisabled =
-    !activeMachineId &&
-    processingCount === 0 &&
-    actionableMachineIDs.length !== machinesToAction.length;
+  const actionStatus = ACTIONS.find(({ name }) => name === action)?.status;
+  const statuses = useSelector(machineSelectors.statuses);
+  // The form expects one error, so we only show the latest error with the
+  // assumption that all selected machines fail in the same way.
+  const errors = useSelector((state: RootState) =>
+    machineSelectors.eventErrorsForIds(
+      state,
+      machines.map(({ system_id }) => system_id),
+      kebabToCamelCase(action)
+    )
+  )[0]?.error;
+  const processingCount = getProcessingCount(machines, statuses, actionStatus);
+  const [actionStarted] = useCycled(processingCount !== 0);
+  const actionableMachineIDs = machines.reduce<Machine[MachineMeta.PK][]>(
+    (machineIDs, machine) =>
+      canOpenActionForm(machine, action)
+        ? [...machineIDs, machine.system_id]
+        : machineIDs,
+    []
+  );
+  // Show a warning if not all the selected machines can perform the selected
+  // action, unless an action has already been started in which case we want to
+  // maintain the form being rendered.
+  const showWarning =
+    !viewingDetails &&
+    !actionStarted &&
```

Comment:
Instead of checking that `processingCount === 0`, this now just checks whether an action has started or not. This fixes a bug where if some, but not all, machines errored the form would close before you could see the error message.